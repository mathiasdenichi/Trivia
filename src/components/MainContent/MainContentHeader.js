import React from 'react'
import PropTypes from 'prop-types'

const Header = ({ category, difficulty }) => (
  <React.Fragment>
    <h1 id='category' key={`${category}h1`}>{category}</h1>
    <h2 id='difficulty' key={`${category}h2`}>Difficulty: {difficulty}</h2>
  </React.Fragment>
)

Header.propTypes = {
  category: PropTypes.string,
  difficulty: PropTypes.string,
}

Header.defaultProps = {
  category: '',
  difficulty: '',
}

export default Header
