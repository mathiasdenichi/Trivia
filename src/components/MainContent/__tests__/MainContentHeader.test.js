import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import MainContentHeader from '../MainContentHeader'


configure({ adapter: new Adapter() })

describe('Main Content Header', () => {
  const wrapper = shallow(<MainContentHeader />)
  it('renders the component and matches it to snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
