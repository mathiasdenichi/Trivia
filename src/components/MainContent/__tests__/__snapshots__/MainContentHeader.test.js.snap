// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Content Header renders the component and matches it to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    category=""
    difficulty=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1
          id="category"
        >
          
        </h1>,
        <h2
          id="difficulty"
        >
          Difficulty: 
          
        </h2>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "h1",
        "nodeType": "host",
        "props": Object {
          "children": "",
          "id": "category",
        },
        "ref": null,
        "rendered": "",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": "h2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Difficulty: ",
            "",
          ],
          "id": "difficulty",
        },
        "ref": null,
        "rendered": Array [
          "Difficulty: ",
          "",
        ],
        "type": "h2",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1
            id="category"
          >
            
          </h1>,
          <h2
            id="difficulty"
          >
            Difficulty: 
            
          </h2>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "h1",
          "nodeType": "host",
          "props": Object {
            "children": "",
            "id": "category",
          },
          "ref": null,
          "rendered": "",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": "h2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Difficulty: ",
              "",
            ],
            "id": "difficulty",
          },
          "ref": null,
          "rendered": Array [
            "Difficulty: ",
            "",
          ],
          "type": "h2",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
